torch/csrc/autograd/function.h

```diff
@@ -97,6 +107,13 @@ struct TORCH_API Node : std::enable_shared_from_this<Node> {
       next_edges_(std::move(next_edges)) {
     if (AnomalyMode::is_enabled()) {
       metadata()->store_stack();
+
+      // If anomaly mode is enabled and graph is constructed, then assign this
+      // node to a parent. A parent is a Node where this Node is created.
+      // We are tracking the parents to track multiple backward operations.
+      if (GradMode::is_enabled()) {
+        assign_parent();
```

Comment:
Not sure if this indirection is really needed, you could just replace this line with `metadata()->assign_parent(current_evaluating_node);` since you already access the metadata on the line above.

Reply:
Done this by moving everything I wrote from `function.cpp` to `function.h`